lib/rules/no-restricted-imports.js

```diff
@@ -121,31 +125,67 @@ module.exports = {
         }
 
         /**
+         * Report a restricted path specifically for patterns.
+         * @param {node} node representing the restricted path reference
+         * @returns {void}
+         * @private
+         */
+        function reportPathForPatterns(node) {
+            const importName = node.source.value.trim();
+
+            context.report({
+                node,
+                message: "'{{importName}}' import is restricted from being used by a pattern.",
+                data: {
+                    importName
+                }
+            });
+        }
+
+        /**
          * Check if the given name is a restricted path name.
-         * @param {string} name name of a variable
+         * @param {string} name - path of the import
+         * @param {[string]} [objectNames] - array of object names that are being imported
          * @returns {boolean} whether the variable is a restricted path or not
          * @private
          */
-        function isRestrictedPath(name) {
-            return Object.prototype.hasOwnProperty.call(restrictedPathMessages, name);
+        function isRestrictedPath(name, objectNames) {
+            let isRestricted = false;
+
+            if (Object.prototype.hasOwnProperty.call(restrictedPathMessages, name)) {
+                if (restrictedPathMessages[name].objectName) {
+                    isRestricted = objectNames.indexOf(restrictedPathMessages[name].objectName) !== -1;
+                } else {
+                    isRestricted = true;
+                }
+            }
+
+            return isRestricted;
+        }
+
+        /**
+         * Check if the given name is a restricted path name.
+         * @param {string} name - path of the import
+         * @returns {boolean} whether the variable is a restricted pattern or not
+         * @private
+         */
+        function isRestrictedPattern(name) {
+            return restrictedPatterns.length > 0 && restrictedPatternsMatcher.ignores(name);
         }
 
         return {
             ImportDeclaration(node) {
                 if (node && node.source && node.source.value) {
                     const importName = node.source.value.trim();
+                    const importObjectNames = node.specifiers.map(specifier => (
+                        (specifier && specifier.local && specifier.local.name) || null
```

Comment:
I don't think you want to use `local` here. I think you want `imported` instead. Since the way your code is written right now, it will check the local variable name, not exported variable name (which is most likely what you want).

Reply:
I'll give this a shot, thanks.

Reply:
Good catch.  `specifier.imported` is populated for the use case when you are renaming imports.  For the use cases I was testing, `local` is the only field populated (so I will continue using that as a fallback when `imported` is not present on `specifier` object.